### PR TITLE
fix(DB/Event): Love is in the Air event missing Love Boat vehicles 1

### PR DESCRIPTION
**Pull request by sogladev**
## Changes Proposed:

This PR proposes changes to:
-  [ ] Core (units, players, creatures, game systems).
-  [ ] Scripts (bosses, spell scripts, creature scripts).
-  [x] Database (SAI, creatures, etc).

Adds Love Boats to Love is in the Air event

- [x] boats that spawn in Darnassus, Undercity, Stormwind
- [x] right-click to board
- [x] passengers receive love buff when 2 characters are boarded
- [x] boats follow waypoints
- [ ] at the end of the waypoint, passengers are ejected and exit on land

## Issues Addressed:

- Progresses https://github.com/azerothcore/azerothcore-wotlk/issues/19360

## SOURCE:

The changes have been validated through:
- [ ] Live research (checked on live servers, e.g Classic WotLK, Retail, etc.)
- [ ] Sniffs (remember to share them with the open source community!)
- [ ] Video evidence, knowledge databases or other public sources (e.g forums, Wowhead, etc.)
- [x] The changes promoted by this pull request come partially or entirely from another project (cherry-pick). **Cherry-picks must be committed using the proper --author tag in order to be accepted, thus crediting the original authors, unless otherwise unable to be found**

https://github.com/TrinityCore/TrinityCore/commit/23961001b5b14d9672dd2c32293f6d75f615caeb

## Tests Performed:

This PR has been:
- [x] Tested in-game by the author.
- [ ] Tested in-game by other community members/someone else other than the author/has been live on production servers.
- [ ] This pull request requires further testing and may have edge cases to be tested.

## How to Test the Changes:

- [x] This pull request can be tested by following the reproduction steps provided in the linked issue
- [x] This pull request requires further testing. Provide steps to test your changes. If it requires any specific setup e.g multiple players please specify it as well.

```
.event start 8
```
go to lake/river/canal in Darnassus, Undercity, Stormwind

tp to boat summoners
```
.go c id 36817
.go c id 37964
.go c id 37981
```

## Known Issues and TODO List:

- [ ] Passengers are not correctly ejected at the end of the waypoint. Passengers should exit on land. ACore is missing a `vehicle_seat_addon` ([introduced in TC335 here](https://github.com/TrinityCore/TrinityCore/commit/98d6c501d7c1a7a632c6ff8b1d46c7d0d4ae5b37)) table to specify vehicle exit positions. hard-coded exit positions are possible to vehicles (615 616 548) seats (7326 7327 7328 7329 6446 6447)
- [ ] https://github.com/azerothcore/azerothcore-wotlk/issues/11159

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by heyitsbench** [comment]
Would like to double check the waypoints against sniffs if/when I get a chance, otherwise good find. 👍

File: data/sql/updates/pending_db_world/rev_1725993194571320983.sql [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+(36812, 0, 1, 2, 40, 0, 100, 0, 21, 36812, 0, 0, 0, 0, 11, 50630, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 'Stormwind Love Boat - On Point 21 of Path 36812 Reached - Cast \'Eject All Passengers\''),
+(36812, 0, 2, 0, 61, 0, 100, 0, 0, 0, 0, 0, 0, 0, 41, 2000, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 'Stormwind Love Boat - On Point 21 of Path 36812 Reached - Despawn In 2000 ms'),
+(36812, 0, 3, 4, 54, 0, 100, 0, 0, 0, 0, 0, 0, 0, 11, 70143, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 'Stormwind Love Boat - On Just Summoned - Cast \'Vehicle Test\''),
+(36812, 0, 4, 0, 61, 0, 100, 0, 0, 0, 0, 0, 0, 0, 48, 1, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 'Stormwind Love Boat - On Just Summoned - Set Active On'),
+(36812, 0, 5, 0, 1, 0, 100, 0, 0, 0, 3000, 3000, 0, 0, 75, 69341, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 'Stormwind Love Boat - Out of Combat - Add Aura \'Love Boat\''),
+(36812, 0, 6, 0, 23, 0, 100, 0, 69342, 2, 1000, 1000, 0, 0, 75, 62011, 0, 0, 0, 0, 0, 17, 0, 20, 0, 0, 0, 0, 0, 0, 'Stormwind Love Boat - On Aura \'Ride Vehicle\' - Add Aura \'Cosmetic - Love\''),
```
Comment (heyitsbench) [inline data/sql/updates/pending_db_world/rev_1725993194571320983.sql:33]: Does this happen to any passenger that rides the boat? If so, that's incorrect, as this aura only gets applied when two passengers ride the boat at the same time.

**Comment by sogladev** [comment]
updated the aura id. Also an issue appeared. When someone boards their position isn't updated to other player's :laughing:  I didn't notice earlier because I was using the patch from https://github.com/azerothcore/azerothcore-wotlk/issues/11159#issuecomment-2336869205  

- https://github.com/azerothcore/azerothcore-wotlk/issues/11159

**Comment by sudlud** [comment]
> updated the aura id. Also an issue appeared. When someone boards their position isn't updated to other player's 😆 I didn't notice earlier because I was using the patch from [#11159 (comment)](https://github.com/azerothcore/azerothcore-wotlk/issues/11159#issuecomment-2336869205)
> 
> * [Passenger Bug on Mounts  #11159](https://github.com/azerothcore/azerothcore-wotlk/issues/11159)

so is this PR final for now? and would you mind opening a PR with the patch you mentioned :)

**Comment by sogladev** [comment]
yes, this PR is final

**Comment by sudlud** [comment]
just tested this, somehow there are no boats beeing spawned in Stormwind? Undercity and Darnassus work as expected

File: data/sql/updates/pending_db_world/rev_1725993194571320983.sql [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+(8, @GUID),
+(8, @GUID+1),
+(8, @GUID+2);
+
+UPDATE `creature_template` SET `npcflag`=16777216,`VehicleId`=616,`IconName`='vehichleCursor' WHERE `entry`=37966;
+UPDATE `creature_template` SET `npcflag`=16777216,`VehicleId`=615,`IconName`='vehichleCursor' WHERE `entry`=37980;
+UPDATE `creature_template` SET `npcflag`=16777216,`VehicleId`=548,`IconName`='vehichleCursor' WHERE `entry`=36812;
+
+-- trigger NPC, make invisible to players
+UPDATE `creature_template` SET `flags_extra` = `flags_extra` | 128 WHERE (`entry` IN (36817, 37964, 36812));
```
[the remark this answers is not in the document]
Comment (sudlud) [reply data/sql/updates/pending_db_world/rev_1725993194571320983.sql:21]: you accidantely made the boats invisible :p

**Comment by sudlud** [comment]
can you please also open issue(s) for the open TODO(s) if they do not exist yet

**Review by sudlud — APPROVED** [review]

**Comment by sogladev** [comment]
thanks for fixing my error. I created an issue for the remaining TODO and updated the PR description

**Pull request merged**